src/Orleans/Configuration/Options/StatisticsOptions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using System;
+
+namespace Orleans.Hosting
```

Comment:
By having component configuration poco classes in the hosting namespace, as opposed to the namespace of the component being configured, we are unnecessarily associating all components in the system with how these components are hosted.  This is architecturally backwards.  Technically this works, but IMO is a bad practice.  For discoverability it should be sufficient that the extension functions be in the hosting namespace.  To configure a specific component, say an AzureTableStorageProvider, it is not unreasonable, in fact it's more clear and explicit, to include the namespace of the feature being configured.

Reply:
@jdom ok, what to do then I thought we agree on it during standup...

Reply:
I'm not convinced yet with either approach. If we were to have an OptionsBuilder like suggested in https://github.com/aspnet/Options/issues/237, then perhaps the decision would be easier and similar to what @jason-bragg is saying, because we could have extension methods for configuring each specific option we would like to expose (and most of the times you would add the provider instance at the same time you configure).
If we don't have that, then the decision is more difficult, as we want the `.Configure<TOptions>(...)` be easily discoverable.
IMO, as of today without OptionsBuilder or similar functionality, I would prefer that everything needed to configure the silo or client be in the default Hosting namespace.